src/Nethermind/Nethermind.Blockchain/Processing/BlockchainProcessor.cs

```diff
@@ -297,9 +293,34 @@ private void RunProcessingLoop()
             if (_logger.IsInfo) _logger.Info("Block processor queue stopped.");
         }
 
+        private void FireProcessingQueueEmpty()
+        {
+            if (IsEmpty)
+            {
+                ProcessingQueueEmpty?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
         public event EventHandler ProcessingQueueEmpty;
 
         public bool IsEmpty => _blockQueue.Count == 0 && _recoveryQueue.Count == 0;
+        public int Count => _blockQueue.Count + _recoveryQueue.Count;
+
+        public IEnumerable<Block> Blocks
+        {
+            get
+            {
+                foreach (BlockRef blockRef in _blockQueue)
```

Comment:
what is this? it will consume the queue - will remove blocks from the queue

Reply:
Already cleaned up, was there for debugging